affine/__init__.py

```
@@ -63,8 +63,13 @@ def _trace(self, msg, *args, **kwargs):
 logger = logging.getLogger("affine")
 def setup_logging(verbosity: int):
     if not getattr(setup_logging, "_prom_started", False):
-        try: start_http_server(METRICS_PORT, METRICS_ADDR, registry=REGISTRY)
-        except: pass
+        try: 
+            start_http_server(METRICS_PORT, METRICS_ADDR, registry=REGISTRY)
+        except Exception as e:
+            # SECURITY FIX: Proper error handling instead of bare except
+            # Log the specific error for debugging while allowing startup to continue
+            logger.warning(f"Failed to start metrics server on {METRICS_ADDR}:{METRICS_PORT}: {e}")
+            pass
```

Better than crashing on port conflicts. However, still uses bare pass after logging and doesn't check if port is already in use. Could silently fail and lose all metrics.